CMSgov/cmcs-eregulations
# EREGCSC-1385 -- Section (and Subpart) picker

Resolves [EREGCSC-1384](https://jiraent.cms.gov/browse/EREGCSC-1384) and [EREGCSC-1385](https://jiraent.cms.gov/browse/EREGCSC-1385)

**Description**
Implements Part-view tab behavior in the Part view of the prototype for the Multi-View Nav concept.  See Figma [here](https://www.figma.com/proto/OI3yDAbymYvHU3L7gTkhxR/eRegs-Multi-View-Nav-%26-View-Switching?page-id=2701%3A3601&node-id=2701%3A4567&viewport=-855%2C1864%2C0.07&scaling=scale-down&starting-point-node-id=2701%3A3615).

**This pull request changes:**

- Enables TOC/Part/Subpart/Section tab navigation on the Part view of the prototype
- Adds "split-tab" button functionality to Subpart and Section tabs
   - Extends the existing FancyDropdown component to be used as split tab button with chevron label
   - Clicking the down chevron on the subpart and section tabs will open associated list of subparts or sections
- Refactors Part view's `path` in Vue Router to remove `resourcesDisplay` param and add `tab` optional param to select a tab on load
- Adds `subpart` and `section` query parameters to set seleted subpart and section
   - these query params control the Subpart and Section tab labels, and will ultimately control the content of the Subpart and Section  
- Handles orphan edge cases in some ways (test with 433.1)
> **Note**
> Not all orphan edge cases are handled; there shouldn't be any errors, but it's possible that the Subpart tab label may show a labeled subpart (e.g. "Subpart A") even though the orphan is not in a subpart.

**Steps to manually verify this change:**

1. Visit the JIRA stories linked above to get an idea of the correct tab behavior.
2. Visit prototype deployment [here](https://d2rrgccs0ftamw.cloudfront.net/) and visit any part
3. Click the TOC/Part/Subpart/Tabs and see how they react.
4. Click the split tab button (the chevron) and make sure the subpart and section lists behave as per the tab behavior outlined in the stories above
